### PR TITLE
impl: check for empty EndpointOption in PopulateCommonOptions

### PR DESCRIPTION
**Pull request by scotthart**
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/13190)

### CODE REVIEW

**Review by dbolduc — APPROVED** [review]

File: google/cloud/internal/populate_common_options_test.cc [hunk 1]
```diff
@@ -49,6 +49,62 @@ TEST(PopulateCommonOptions, Simple) {
...
+  EXPECT_THAT(actual.get<AuthorityOption>(), Eq("default"));
+  EXPECT_FALSE(actual.has<UserProjectOption>());
+  EXPECT_TRUE(actual.has<TracingComponentsOption>());
+  EXPECT_THAT(actual.get<TracingComponentsOption>(), IsEmpty());
+  EXPECT_TRUE(actual.has<UserAgentProductsOption>());
+  EXPECT_THAT(actual.get<UserAgentProductsOption>(),
+              Contains(UserAgentPrefix()));
+}
+
+TEST(PopulateCommonOptions, EmptyEndpointEnvVar) {
```
Comment (dbolduc) [inline google/cloud/internal/populate_common_options_test.cc:70]: Actually, I think we can remove this test. I think it is covered by `EndpointAuthority` (not that I love that test. It has very complicated logic for a unit test).

File: google/cloud/internal/populate_common_options_test.cc [hunk 2]
```diff
@@ -49,6 +49,62 @@ TEST(PopulateCommonOptions, Simple) {
...
+  EXPECT_THAT(actual.get<AuthorityOption>(), Eq("default"));
+  EXPECT_FALSE(actual.has<UserProjectOption>());
+  EXPECT_TRUE(actual.has<TracingComponentsOption>());
+  EXPECT_THAT(actual.get<TracingComponentsOption>(), IsEmpty());
+  EXPECT_TRUE(actual.has<UserAgentProductsOption>());
+  EXPECT_THAT(actual.get<UserAgentProductsOption>(),
+              Contains(UserAgentPrefix()));
+}
+
+TEST(PopulateCommonOptions, EmptyEmulatorEnvVar) {
```
Comment (dbolduc) [inline google/cloud/internal/populate_common_options_test.cc:89]: same

**Review by dbolduc — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/13190)* status: 0 of 2 files reviewed, 5 unresolved discussions (waiting on @scotthart)

___
*[`google/cloud/internal/populate_common_options.cc` line 31 at r1](https://reviewable.io/reviews/googleapis/google-cloud-cpp/13190#gh-1406775102:-NkHz9NA9natczAurHoi:bvkmwkz) ([raw file](https://github.com/googleapis/google-cloud-cpp/blob/3cc996e318b2784399145e0e3c545e9058396113/google/cloud/internal/populate_common_options.cc#L31)):*
> I think the expression, as written, is more expressive of intent.

They are the same to me, but checking `!has` first does an extra look up when the option is set. :shrug:

**Review by scotthart — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/13190)* status: 0 of 2 files reviewed, 5 unresolved discussions (waiting on @dbolduc)

___
*[`google/cloud/internal/populate_common_options.cc` line 31 at r1](https://reviewable.io/reviews/googleapis/google-cloud-cpp/13190#gh-1406775102:-NkLyK-D6-CGdj4Ux7Xz:bn5jpss) ([raw file](https://github.com/googleapis/google-cloud-cpp/blob/3cc996e318b2784399145e0e3c545e9058396113/google/cloud/internal/populate_common_options.cc#L31)):*
<details><summary><i>Previously, dbolduc (Darren Bolduc) wrote…</i></summary><blockquote>

> I think the expression, as written, is more expressive of intent.

They are the same to me, but checking `!has` first does an extra look up when the option is set. :shrug:
</blockquote></details>

Per team discussion, going with the non-compound condition.

**Pull request merged**